Tools/TestWebKitAPI/Tests/WTF/NativePromise.cpp

```diff
@@ -0,0 +1,837 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
```

Comment:
This is probably an incorrect copyright header.  I think it should involve Mozilla.

Reply:
I have fully rewritten the code multiple times from scratch because MozPromise is under ML2.0 which we can’t use currently. 

I can mention the origin of the idea however

Reply:
Will add, missed that you were referring to the tests.